scripts/experiment_2/nbconverted/4_similarity_analysis_pca_svcca.py

```diff
@@ -0,0 +1,169 @@
+
+
+# In[8]:
+
+
+# Convert output to pandas dataframe
+svcca_pca_df = pd.DataFrame(output_list, columns=["svcca_mean_similarity"], index=num_batches)
+svcca_pca_df
+
+
+# In[9]:
+
+
+get_ipython().run_cell_magic('time', '', '# Permute simulated data\nshuffled_simulated_arr = []\nnum_samples = simulated_data.shape[0]\n\nfor i in range(num_samples):\n    row = list(simulated_data.values[i])\n    shuffled_simulated_row = random.sample(row, len(row))\n    shuffled_simulated_arr.append(shuffled_simulated_row)\n\nshuffled_simulated_data = pd.DataFrame(shuffled_simulated_arr, \n                                       index=simulated_data.index, \n                                       columns=simulated_data.columns)\nshuffled_simulated_data.head()')
+
+
+# In[10]:
+
+
+get_ipython().run_cell_magic('time', '', '# PCA\nsimulated_data_PCAencoded = pca.fit_transform(simulated_data)\n\n\nsimulated_data_PCAencoded_df = pd.DataFrame(simulated_data_PCAencoded,\n                                     index=simulated_data.index\n                                     )\n\nshuffled_data_PCAencoded = pca.fit_transform(shuffled_simulated_data)\n\n\nshuffled_data_PCAencoded_df = pd.DataFrame(shuffled_data_PCAencoded,\n                                     index=shuffled_simulated_data.index\n                                     )\n \n# SVCCA\nsvcca_results = cca_core.get_cca_similarity(simulated_data_PCAencoded_df.T,\n                                      shuffled_data_PCAencoded_df.T,\n                                      verbose=False)\n\npermuted_svcca = np.mean(svcca_results["cca_coef1"])\nprint(permuted_svcca)')
+
+
+# In[11]:
+
+
+# Plot
+threshold = pd.DataFrame(
+    pd.np.tile(
+        permuted_svcca,
+        (len(num_batches), 1)),
+    index=num_batches,
+    columns=['svcca'])
+
+ggplot(svcca_pca_df, aes(x=num_batches, y='svcca_mean_similarity'))     + geom_line()     + geom_line(aes(x=num_batches, y='svcca'), threshold, linetype='dashed')     + xlab('Number of Batch Effects')     + ylab('SVCCA')     + ggtitle('Similarity across increasing batch effects')
```

Comment:
may want to consider log(x-axis) or some transformation since batch_counts>100 seem not useful in these comparisons

Reply:
This is something I can definitely play with in the refactor